common/sagas/schedule/currentSchedulingToggle.ts

```diff
@@ -0,0 +1,28 @@
+import { setGasLimitField } from 'actions/transaction/actionCreators/fields';
+import { call, put, takeLatest } from 'redux-saga/effects';
+import { SagaIterator } from 'redux-saga';
+import { TypeKeys } from 'actions/schedule/constants';
+import { SetGasLimitFieldAction } from 'actions/transaction';
+import { EAC_SCHEDULING_CONFIG } from 'libs/scheduling';
+import BN from 'bn.js';
+import { SetSchedulingToggleAction } from 'actions/schedule/actionTypes';
+
+export function* setGasLimitForScheduling({
+  payload: { value: useScheduling }
+}: SetSchedulingToggleAction): SagaIterator {
+  const gasLimit = useScheduling ? EAC_SCHEDULING_CONFIG.SCHEDULING_GAS_LIMIT : new BN('21000');
+
+  yield call(setGasLimit, {
+    raw: gasLimit.toString(),
+    value: gasLimit
+  });
+}
+
+export function* setGasLimit(payload: SetGasLimitFieldAction['payload']) {
```

Comment:
This function isn't used anywhere externally, would rather just have inline yield put(setGasLimitField(payload)); in the two references above to avoid extra indirection